src/terminal/parser/stateMachine.cpp

```diff
@@ -2161,11 +2159,12 @@ template<typename TLambda>
 bool StateMachine::_SafeExecute(TLambda&& lambda)
 try
 {
-    return lambda();
-}
-catch (const ShutdownException&)
-{
-    throw;
+    bool ok = lambda();
+    if (!ok)
+    {
+        FlushToTerminal();
+    }

```

Comment:
This is the important bit.

Reply:
Should we flush it even if it made us throw an exception?

Reply:
I think that's an edge case that's not too important, at least initially. There shouldn't be any exceptions unless it's an OOM or similar exceptional issue. Failure to parse a sequence should never raise an exception.